include/nuttx/video/rfb.h

```diff
@@ -356,7 +356,7 @@ struct rfb_setencodings_s
 };
 
 #define SIZEOF_RFB_SETENCODINGS_S(n) \
-  (sizeof(struct rfb_setencodings_s) + (((n) - 1) << 2))
+  (sizeof(struct rfb_setencodings_s) + (n == 0 ? -4 : (n - 1) << 2))
```

Comment:
Just as an alternative, but will require `#include <nuttx/compiler.h>`
```suggestion
  (offsetof(struct rfb_setencodings_s, encodings) + ((n) << 2))
```